packages/swing-store/src/snapStore.js

```diff
@@ -106,41 +134,46 @@ export function makeSnapStore(
   }
 
   /**
-   * @param {string} dest basename, relative to root
-   * @param { (name: string) => Promise<void> } thunk
-   * @returns { Promise<{ filename: string, size: number }> }
+   * Creates a file atomically by moving in place a temp file
+   * populated by a callback.
+   *
+   * @param {string} destFilePath absolute path of file to be written
+   * @param { (name: string) => Promise<void> } writeContents
+   * @returns {Promise<void>}
    */
-  async function atomicWrite(dest, thunk) {
-    assert(!dest.includes('/'));
-    const tmp = await ptmpName({ tmpdir: root, template: 'atomic-XXXXXX' });
-    let result;
+  async function atomicWrite(destFilePath, writeContents) {
+    // Atomicity requires remaining on the same filesystem,
+    // so we perform all operations in the destination directory.
+    const dir = resolve(destFilePath, '..');
```

Comment:
Why not use `dirname`?

```suggestion
    const dir = path.dirname(destFilePath);
```

Reply:
`resolve` is an endowment to `makeSnapStore`, and I didn't want to require another one or risk conflict between it and the built-in path module (however unlikely that may be). But I've restored root-relative behavior to avoid that, and this should be refactored in the performance-improving followup anyway.